inc/classes/subscriber/Tools/class-cloudflare-subscriber.php

```diff
@@ -0,0 +1,888 @@
+<?php
+namespace WP_Rocket\Subscriber\Tools;
+
+use WP_Rocket\Admin\Options_Data as Options;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+
+/**
+ * Cloudflare Subscriber.
+ *
+ * @since  3.5
+ * @author Soponar Cristina
+ */
+class Cloudflare_Subscriber implements Subscriber_Interface {
+
+	/**
+	 * Instance of the Option_Data class.
+	 *
+	 * @var    Options
+	 * @since  3.5
+	 * @access private
+	 * @author Soponar Cristina
+	 */
+	private $options;
+
+	/**
+	 * Cloudflare User Agent
+	 */
+	const CF_USER_AGENT = 'wp-rocket/' . WP_ROCKET_VERSION;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since  3.5
+	 * @access public
+	 * @author Soponar Cristina
+	 *
+	 * @param Options $options Instance of the Option_Data class.
+	 */
+	public function __construct( Options $options) {
+		$this->options = $options;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_subscribed_events() {
+		if ( ! get_rocket_option( 'do_cloudflare' ) ) {
```

Comment:
I believe it would be better to do this check via a private method like `is_enabled()` called in each methods hooked instead of the method used to subscribed the events, to improve the separation of concerns, and also remove coupling with procedural code